File: 02_Supervised_Learning-Regression/Python/Code/03_simple_linear_regression.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Tue Nov 22 21:32:06 2016

@author: pavan.govindraj
"""

#simple linear regression
# Data preprocessing => import the dataset
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Salary_Data.csv')

X = dataset.iloc[:, :-1].values  #independent variable/matric
Y = dataset.iloc[:, 1].values    #dependent variable/vector


#Splitting the dataset in to training set and test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,Y,test_size = 1/3, 
                                                    random_state = 0)

#Feature scaling
"""
#May be required for some dataset 
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
"""

#Fitting simple linear regression to the training set 
#linear equation Y= a *X + b. where Y -> Dependent Variable; a -> slope
#b -> intercept ; X -> Independent Variable
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)
#regressor.score(X_train, y_train)

#Equation coefficient and Intercept
print('Coefficient: ', regressor.coef_)
print('Intercept: ', regressor.intercept_)

#Predicting the Test set results
y_pred = regressor.predict(X_test)

#Visualising the Training set results
plt.scatter(X_train, y_train, color = 'red')
plt.plot(X_train,regressor.predict(X_train), color = 'blue' )
plt.title("Salary VS. Experience (Training set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.show()

#Visualising the Test set results
plt.scatter(X_test, y_test, color = 'red')
plt.plot(X_train,regressor.predict(X_train), color = 'blue' )
plt.title("Salary VS. Experience (Test set)")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.show()

File: 02_Supervised_Learning-Regression/R/Code/03_simple_linear_regression.R
#Data Preprocessing ; importing the

dataset = read.csv('Salary_Data.csv')
#dataset = dataset[, 2:3]

#Splitting the dataset in to training set and test set
#install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3 )
training_set = subset(dataset, split == TRUE)
test_set= subset(dataset, split == FALSE)

#Feature scaling
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])

#Fitting simple linear regression to the Training set
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)
#summary(regressor) - in console to see the relavent info about regressor

#predicting the test set results
y_pred = predict(regressor, test_set)

#Visualising the Training set results
#install.packages('ggplot2')
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience , y = training_set$Salary),
             color = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor,
                                                              newdata = training_set)),
                color = 'blue') +
  ggtitle("Salary Vs. Experience (Training set)") +
  xlab("YearsExperiece") +
  ylab("Salary")

#Visualising the Test set results
ggplot() +
  geom_point(aes(x = test_set$YearsExperience , y = test_set$Salary),
             color = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor,
                                                              newdata = training_set)),
            color = 'blue') +
  ggtitle("Salary Vs. Experience (Training set)") +
  xlab("YearsExperiece") +
  ylab("Salary")



